routes/api.php

```diff
@@ -51,4 +51,7 @@
     // actions
     $this->get('actions', 'ActionsController@index');
     $this->get('actions/{action}', 'ActionsController@show');
+
+    // action stats
+    $this->get('action-stats', 'ActionStatsController@index');
```

Comment:
This is such an odd one to think about, since it's really stats about the action + school together! I wonder if we'd want to explicitly name it as `action-school-stats` (since it's not really applicable for non-school actions)? Or nest it under a placeholder "schools" resource, like `/actions/:action_id/schools/:school_id/stats`?

Reply:
My thought was to name it exactly per model for simplicity sake (once this is in place, I'll be starting on adding this into GraphQL).

I went with `ActionStat` vs using the word `School` somewhere in the name -- because in design meetings there had been talk of eventually renaming `school_id` as something like `affiliation_id`, e.g. to support youth groups, after school groups, etc. But because we're not there yet, it seemed a lot clearer to store these as `school_id`.

I do like the idea of defining the `school_id` and `action_id` via URL params instead of query params... but it potentially adds a lot overhead that I thought might not be necessary. But if we went this route (and still assuming `ActionStat` as the model name), it could look like this?

`GET /actions/:action_id/stats` - This is our leaderboard for an action. GraphQL query could be named `Action.getStats(options)`

`GET /actions/:action_id/schools/:school_id/stats` - This is the school's stat for the action, which we'll display to user upon selecting a school in a School Finder (we'll need to add a new `actionId` parameter to the `additionalContent` in a `SectionBlock` to know which `ActionStat` to display). GraphQL query could be `School.getStatsForActionId()`

`GET /schools/:school_id/stats` - All accepted actions a school has taken, GraphQL `School.getStats(options)`

Reply:
Thinking more about the GraphQL methods makes me think we could get away with just a single index route, because we're likely never going to use this endpoint anywhere else. It's also got me leaning more towards just calling the model `ActionSchoolStat` or maybe potentially better or worse `SchoolAction` for have a single index endpoint (e.g. `action-school-stats` like you suggested)

Reply:
Ahh, that makes sense re: the naming & potential renames in the pipeline. I can also see the argument that `?filter[…]=` is easier to deal with that URL-params in our resolvers. I'm good to leave it as is & we can always see if something starts to make more sense as it solidifies!

> `GET /actions/:action_id/stats` - This is our leaderboard for an action. GraphQL query could be named `Action.getStats(options)`
> 
> `GET /schools/:school_id/stats` - All accepted actions a school has taken, GraphQL `School.getStats(options)`

Nitpick, but I think I'd recommend naming these as `stats` or `statistics` rather than `getStats` (so it's a "thing" in the schema, rather than an imperative request).

> `GET /actions/:action_id/schools/:school_id/stats` - This is the school's stat for the action, which we'll display to user upon selecting a school in a School Finder (we'll need to add a new `actionId` parameter to the `additionalContent` in a `SectionBlock` to know which `ActionStat` to display). GraphQL query could be `School.getStatsForActionId()`

Again, sorta nitpicky but this could be an optional argument on `School.statistics`, so we could do something like this:

```gql
{
  school(…) {
    name
    stats { # <-- no argument, so we leave `?filter[action_id]=` out!
      actionId
      acceptedQuantity
    }
    thisAction: stats(actionId: 934) { # <-- ?filter[action_id]=934
      acceptedQuantity
    }
}
```